### PR TITLE
Fixed alter data commands

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?

More fixes related to #1833 
Rebased on #1833 

#### What's this PR do?

The refactored edX data (#1806) broke our seed data functionality. #1833 fixes up the seed data functionality, and this PR fixes up the alter_data management commands specifically

#### How should this be manually tested?

With seed data loaded, run `./manage.py alter_data --usage` and/or `./manage.py alter_data --help` for usage details. Run through a few of the commands and check the /dashboard page for the Analog Program to make sure the results are what you expect

### CODE REVIEW

File: seed_data/management/commands/unseed_db.py [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+    """
+    table_names = [model._meta.db_table for model in models]
+    with connection.cursor() as cursor:
+        for table_name in table_names:
+            cursor.execute("DROP RULE delete_protect ON {}".format(table_name))
+        try:
+            yield
+        finally:
+            table_names.reverse()
+            for table_name in table_names:
```
Comment (singingwolfboy) [inline seed_data/management/commands/unseed_db.py]: Nitpick: you could just do `for table_name in reversed(table_names):`

File: seed_data/management/commands/seed_db.py [hunk 2]
```diff
@@ -289,10 +294,12 @@ def handle(self, *args, **options):
...
                 fake_programs = deserialize_program_data_list(program_data_list)
-                fake_course_runs = CourseRun.objects.filter(
-                    course__program__description__contains=FAKE_PROGRAM_DESC_PREFIX
-                ).all()
-                fake_user_count = deserialize_user_data_list(user_data_list, fake_course_runs)
+                fake_user_count = deserialize_user_data_list(user_data_list, fake_programs)
+
+            # Handle FA programs
+            fake_financial_aid_programs = filter(lambda program: program.financial_aid_availability, fake_programs)
+            create_tiers(fake_financial_aid_programs, int(options["tiers"]))
```
Comment (singingwolfboy) [inline seed_data/management/commands/seed_db.py]: If `seed_db` now runs the `create_tiers` command as well, could you update the `README` file to indicate that running `manage.py create_tiers` is no longer necessary?

**Comment by singingwolfboy** [comment]
I just tried running the first command in the `--usage` list, and got an exception right away. I'm not sure if it's related to this change, but can you take a look?
```
➜  micromasters git:(alter_data_command_fixes) docker-compose run web ./manage.py alter_data --action=set_course_to_enrolled --username=staff --program-title='Digital' --course-title='100'
[2016-11-30 16:05:29] DEBUG 1 [django.db.backends] utils.py:90 - [d28be4ca82da] - (0.001) SELECT (1) AS "a" FROM "wagtailimages_rendition" WHERE "wagtailimages_rendition"."filter_spec" = '' LIMIT 1; args=('',)
[2016-11-30 16:05:29] DEBUG 1 [django.db.backends] utils.py:90 - [d28be4ca82da] - (0.001) SELECT "auth_user"."id", "auth_user"."password", "auth_user"."last_login", "auth_user"."is_superuser", "auth_user"."username", "auth_user"."first_name", "auth_user"."last_name", "auth_user"."email", "auth_user"."is_staff", "auth_user"."is_active", "auth_user"."date_joined" FROM "auth_user" WHERE "auth_user"."username"::text LIKE '%staff%'; args=('%staff%',)
Traceback (most recent call last):
  File "./manage.py", line 14, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/src/seed_data/management/commands/alter_data.py", line 335, in handle
    user = UserFinder().find(**user_finder_params)
  File "/src/seed_data/lib.py", line 78, in find
    return self.result(users, params)
  File "/src/seed_data/lib.py", line 61, in result
    '\n'.join(str(obj) for obj in objects)
Exception: Multiple User records found with the given params ({'username__contains': 'staff'}).
staff
staffd2027f8adfef44e7
```

File: seed_data/lib.py [hunk 3]
```diff
@@ -275,6 +263,18 @@ def set_or_create(self, *args, **kwargs):
...
 
 
+def ensure_cached_data_freshness(user):
+    """
+    Ensure that all cached edX data will be considered 'fresh' for a User
+    """
+    future = future_date()
+    # Create a dict with all supported cache types mapped to the future timestamp value
+    timestamps = (future, ) * len(CachedEdxUserData.SUPPORTED_CACHES)
+    updated_values = dict(zip(CachedEdxUserData.SUPPORTED_CACHES, timestamps))
```
Comment (gsidebo) [inline seed_data/lib.py]: 👍  i knew this one felt silly

File: seed_data/management/commands/__init__.py [hunk 4]
```diff
@@ -11,3 +11,4 @@
 DEFAULT_GRADE = Decimal('0.75')
 DEFAULT_FAILED_GRADE = Decimal('0.50')
 DEFAULT_COURSE_PRICE = Decimal('123.45')
+PASSED_GRADE = Decimal('0.60')
```
Comment (gsidebo) [inline seed_data/management/commands/__init__.py]: 👍

File: seed_data/management/commands/seed_db.py (hunk 2 above)
Comment (gsidebo) [inline seed_data/management/commands/seed_db.py]: 👍

File: seed_data/management/commands/unseed_db.py (hunk 1 above)
Comment (gsidebo) [inline seed_data/management/commands/unseed_db.py]: 👍

**Comment by gsidebo** [comment]
@singingwolfboy sorry, i should change the `--usage` details to mention that `--username` or `--email` should match the exact user that you want to alter data for. if the user that you normally log into local MM has an email like, say, 'dbaumgold@mit.edu', then you should use `--email=dbaumgold` instead of `--username=staff`. you'll get that 'multiple found' error if your `username` or `email` terms match multiple users

**Comment by gsidebo** [comment]
@singingwolfboy ready for review again

**Pull request merged**